Fail if more than one OIDC JWT credential key property is configured

sberyozkin opened the pull request:
Fixes #42920.

This PR throws a `ConfigurationException` if more than one OIDC Credential JWT key property is configured as the consequences can be very confusing as shown in #42920.

OIDC Credential JWT group is used to configure various properties related to preparing a JWT authentication token which in turn is used to acquire an access token. The type of the key material can affect the way the JWT token is signed and can in turn impact how the OIDC provider handles it, which is why it is important to make sure only a single key property is used.

Other groups must also be reviewed, but I'll start with the JWT group.

Keeping in draft until I add a test

sberyozkin committed:
Fail if more than one OIDC JWT credential key property is configured

sberyozkin commented:
Added a simple test to assert a failure is thrown if more than one JWT signing key or secret property is configured, was not 100% sure if a neater Java expression could be used though :-)

gastaldi commented on extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java:
```diff
@@ -284,8 +284,23 @@ public static boolean isClientSecretBasicAuthRequired(Credentials creds) {
+            keyMaterialConfigCount++;
+        }
+        if (keyMaterialConfigCount > 1) {
+            throw new ConfigurationException("Only a single OIDC JWT credential key property can be configured");
```
This error is cryptic to me. Maybe specify the extra key properties already configured that are not mutually exclusive?

sberyozkin replied:
Hi @gastaldi Hopefully it looks better now, thanks :-)

sberyozkin replied:
@gastaldi Actually, sorry, I'll need to tweak it further, to correctly report `quarkus.oidc-client` vs `quarkus.oidc`, sorry

gastaldi replied:
No problem, I'll keep it approved as I trust you ;)

sberyozkin commented:
Hi @gastaldi So now I've properly fixed it, I had to pass a boolean flag from `quarkus-oidc` and `quarkus-oidc-client` down to `OidcCommonUtils` to correctly determine the prefix. I also removed one redundant check there (from within the Apple POST JWT check function) since both at the `quarkus-oidc` and `quarkus-oidc-client` levels the Apple POST-JWT check is only done if the check I removed from the Apple POST JWT check already succeeded.
Sorry about it, I did not expect this PR be so tough for me :-)
